File: examples/Vulnerable.java
// Deliberately vulnerable Java sample, for testing XSEC.
// Run: xsec scan examples/Vulnerable.java

import java.io.ObjectInputStream;
import java.security.MessageDigest;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;

public class Vulnerable {

    static final String API_KEY = "super_secret_value_123";  // fake, for testing

    void runCommand(String userInput) throws Exception {
        Runtime.getRuntime().exec("ping " + userInput);  // command injection
    }

    ResultSet lookup(Connection conn, String name) throws Exception {
        Statement st = conn.createStatement();
        return st.executeQuery("SELECT * FROM users WHERE name = '" + name + "'");  // SQLi
    }

    Object load(ObjectInputStream in) throws Exception {
        return in.readObject();  // unsafe deserialization (stream created elsewhere)
    }

    Object loadDirect(java.io.InputStream raw) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(raw);  // unsafe deserialization
        return ois.readObject();
    }

    byte[] hash(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");  // weak hash
        return md.digest(data);
    }
}

File: examples/vulnerable.js
// Deliberately vulnerable JS sample, for testing XSEC.
// Run: xsec scan examples/vulnerable.js

const cp = require("child_process");
const crypto = require("crypto");

const API_KEY = "super_secret_value_123";  // fake value, just for testing

function calc(expr) {
  return eval(expr);  // arbitrary code execution
}

function run(userInput) {
  cp.exec(`ls ${userInput}`);  // command injection
}

function render(el, userInput) {
  el.innerHTML = userInput;  // DOM XSS
}

function fingerprint(data) {
  return crypto.createHash("md5").update(data).digest("hex");  // weak hash
}

const agent = { rejectUnauthorized: false };  // TLS verification off

module.exports = { calc, run, render, fingerprint, agent, API_KEY };
